packages/grid/x-data-grid/src/DataGrid/useDataGridComponent.tsx

```diff
@@ -64,6 +63,7 @@ export const useDataGridComponent = (props: DataGridProcessedProps) => {
   useGridInitializeState(rowsMetaStateInitializer, apiRef, props);
   useGridInitializeState(columnMenuStateInitializer, apiRef, props);
 
+  useGridKeyboardNavigation(apiRef, props);
```

Comment:
`useGridSelection` must consume the `cellKeydown` after the navigation have been applied.